polaris-react/src/components/IndexFilters/components/SortButton/SortButton.tsx

```diff
@@ -0,0 +1,153 @@
+import {Popover} from '../../../Popover';
+import {ChoiceList} from '../../../ChoiceList';
+import type {ChoiceListProps} from '../../../ChoiceList';
+import {Tooltip} from '../../../Tooltip';
+import {Box} from '../../../Box';
+import type {SortButtonChoice} from '../../types';
+import {FilterButton} from '../FilterButton';
+
+import {OrderButton} from './components';
+
+export enum SortButtonDirection {
+  Asc = 'asc',
+  Desc = 'desc',
+}
+
+export interface SortButtonProps {
+  choices: SortButtonChoice[];
+  selected: ChoiceListProps['selected'];
+  onChange: (selected: string[]) => void;
+  disabled?: boolean;
+  onChangeKey?: (key: string) => void;
+  onChangeDirection?: (direction: string) => void;
+}
+
+export function SortButton({
+  choices,
+  selected,
+  onChange,
+  disabled,
+  onChangeKey,
+  onChangeDirection,
+}: SortButtonProps) {
+  const i18n = useI18n();
+  const [active, setActive] = useState(false);
+  const [selectedValueKey, selectedDirection] = selected[0].split(' ');
+
+  function handleClick() {
+    setActive((pastActive) => !pastActive);
+  }
+
+  function handleClose() {
+    setActive(false);
+  }
+
+  function handleChangeChoiceList(sel: string[]) {
+    if (onChangeKey) {
+      const [key] = sel[0].split(' ');
+      onChangeKey(key);
+    } else {
+      onChange(sel);
+    }
+  }
+
+  function handleChangeDirection(sel: string[]) {
+    if (onChangeDirection) {
+      const [, direction] = sel[0].split(' ');
+      onChangeDirection(direction);
+    } else {
+      onChange(sel);
+    }
+  }
+
+  const choiceListChoices = useMemo(() => {
+    const choiceCategories = choices.reduce(
+      (acc: ChoiceListProps['choices'], curr) => {
+        const alreadyExists = acc.some((option) => option.label === curr.label);
+        const [, currentValueDirection] = curr.value.split(' ');
+        const isSameDirection = currentValueDirection === selectedDirection;
+        if (!alreadyExists) {
+          return [...acc, curr];
+        }
+        if (isSameDirection) {
+          return acc.map((option) => {
+            if (option.label === curr.label) {
+              return curr;
+            }
+            return option;
+          });
+        }
+        return acc;
+      },
+      [],
+    );
+    return choiceCategories;
+  }, [choices, selectedDirection]);
+
+  const selectedChoices = choices.filter((choice) => {
+    const [currentKey] = choice.value.split(' ');
+    return currentKey === selectedValueKey;
+  });
+
+  const tooltipActivator = (
+    <FilterButton
+      onClick={handleClick}
+      aria-label={i18n.translate('Polaris.IndexFilters.SortButton.ariaLabel')}
+      disabled={disabled}
+    >
+      <Icon source={SortMinor} color="base" />
+    </FilterButton>
+  );
+
+  const activator = (
+    <Tooltip
+      content={i18n.translate('Polaris.IndexFilters.SortButton.tooltip')}
+      preferredPosition="above"
+      hoverDelay={400}
+    >
+      {tooltipActivator}
+    </Tooltip>
+  );
```

Comment:
```suggestion
  const sortButton = (
    <Tooltip
      content={i18n.translate('Polaris.IndexFilters.SortButton.tooltip')}
      preferredPosition="above"
      hoverDelay={400}
    >
      <FilterButton
        onClick={handleClick}
        aria-label={i18n.translate('Polaris.IndexFilters.SortButton.ariaLabel')}
        disabled={disabled}
      >
        <Icon source={SortMinor} color="base" />
      </FilterButton>
    </Tooltip>
  );
```

Since there's multiple activators involved, it'll be helpful for future contributors if we name them for what they render instead of what they activate.